test-cases/gutenberg/writing-flow/rich-text-formatting.md

```diff
@@ -5,82 +5,6 @@
 
 Have a rich-text based component with content (Paragraph, Heading, Quote, Media Caption, etc...)
 
-##### TC001
-
-**Known Issues**
-
-- [Text color selection is not preserved when cursor is placed in the middle of text](https://github.com/WordPress/gutenberg/issues/42714)
-
-**Bold, Italic, strikethrough buttons**
-
-- Press the Bold button and write a bold word.
-- Press the Italic button and write an italic word.
-- Press strikethrough button and write strikethrough text.
-
-**Highlight selected text**
-
-- Write some text and select it.
-- Tap on the Highlight text button (A).
-- Select a color.
-- Expect to see the text with the selected color.
-
-**Highlight text without selection**
-
-- Without selecting any text tap on the highlight text button (A).
-- Select a color and write some text.
-- Expect to see the newly introduced text with the previously selected color.
```

Comment:
I checked the [referenced automated tests](https://github.com/WordPress/gutenberg/pull/49352) and the test cases of [the Paragraph block](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/paragraph/test/edit.native.js) but couldn't find cases related to highlighting text. Hence, probably we should keep these two as manual until we automate them.

Reply:
> For highlighting without a selection, the `typeInRichText` helper does not support inner HTML tags, [only outer HTML tags](https://github.com/WordPress/gutenberg/blob/87bb321d07cceaea4d2013d2468234ec7aaf328e/test/native/integration-test-helpers/rich-text-type.js#L28-L30). Updating it to support inner HTML tags would get complicated really quickly, e.g. how might we handle typing with a text selection that begins within a tag but ends outside of it? Therefore, an e2e test likely makes more sense. I'll reinstate this case.

Good point, although I'm wondering if we need support for inner HTML tags for this case. If we just create a Paragraph block, tap the Highlight text button, and type text, I understand we wouldn't need to place the cursor between inner HTML tags in the assertion. Would this be an option? (I haven't tested this myself, so I might be overlooking an issue about inner HTML tags per your comment).

Reply:
> If we just create a Paragraph block, tap the Highlight text button, and type text, I understand we wouldn't need to place the cursor between inner HTML tags in the assertion. Would this be an option? (I haven't tested this myself, so I might be overlooking an issue about inner HTML tags per your comment).

A `mark` tag is inserted once the Highlight text button is tapped. The next call of `typeInRichText` places the cursor at the end of the input (the expected default). However, the reality is that "the end" is after the closing `mark` tag. To counter this we either need to (A) add support for inner HTML tags or (B) manually adjust for the closing mark tag — e.g. `targetPosition - lengthOfClosingMarkTag` — which feels brittle and cryptic. 

Inner HTML tag support would resolve the issue of difference between rendered string position (position in a visual representation after HTML is parsed and applied) vs literal string position (position in a raw string that includes HTML mark up).

Reply:
> A mark tag is inserted once the Highlight text button is tapped. The next call of typeInRichText places the cursor at the end of the input (the expected default). However, the reality is that "the end" is after the closing mark tag.

Ah, true, I didn't realize that the `mark` tag was inserted upon tapping the button. I understand now the necessity of supporting the inner HTML tag. Thanks David for elaborating 🙇 !

> To counter this we either need to (A) add support for inner HTML tags or (B) manually adjust for the closing mark tag — e.g. `targetPosition - lengthOfClosingMarkTag` — which feels brittle and cryptic.

I agree that option A seems the best way to go 👍 .

> Inner HTML tag support would resolve the issue of difference between rendered string position (position in a visual representation after HTML is parsed and applied) vs literal string position (position in a raw string that includes HTML mark up).

Maybe we could create a follow-up GitHub issue with this, so we can keep track of it. WDYT?

Reply:
I created https://github.com/WordPress/gutenberg/issues/52520 capturing this enhancement.